### ISSUE TITLE
Crazy artifacts on safari

### ISSUE THREAD

**Issue by florian-strasser** [thetarnav/vue-gooey-background#4]
The morphing gooey effect doesnt work on safari :-( Was trying some things, but nothing helped...

You can see the Problem here, Chrome works, but Safari not
http://vue.florian-strasser.de/

**Comment by thetarnav** [comment]
Thanks for the issue, I had no idea since I've never used this with safari. I'll look into it as soon as I get a MacOS working on a VM.
But here is a quote from [CSS Tricks article about the gooey effect](https://css-tricks.com/gooey-effect/#support) I found:
> SVG filters have good support, but not all browsers support them being applied to regular DOM elements, notably Safari.

I will check if this is really the case, but if so, then the best I can do is to find a way to disable this effect only for Safari, to avoid artifacts.

**Comment by florian-strasser** [comment]
I also tried a lot of things i found on stack overflow and yea, it seems like those filter work in safari only on svg elements. I will look into it if i can recreate or modify the script in this way. I was hoping that i just missed an option or something else.

Besides that this is a really cool script, appreciate your affords

When i succeed i will let you know

**Comment by florian-strasser** [comment]
On the other hand there are somehow working examples of this effect with regular DOM Elements:
https://codepen.io/lbebber/pen/OPjxZL

**Comment by florian-strasser** [comment]
In the same article you posted they also talk about space around the element
https://css-tricks.com/gooey-effect/#considerations

Sorry if i write alot, but i'm having so much information while seeking a way to fix this

**Comment by thetarnav** [comment]
That's true, I've also noticed that. But right now I'm having trouble setting up a VM, so I'm not able to iterate on the issue nor even see what kind of artifacts are there :d

**Comment by thetarnav** [comment]
I've applied the same SVG filter that I'm using in the component and shrunk the container element in the codepen that you've sent. https://codepen.io/thetarnav/pen/NWveemN
Of course, it works everywhere I check, but I still need to get my hands on safari somehow.

**Comment by thetarnav** [comment]
Aha! I see.
![image](https://user-images.githubusercontent.com/24491503/142233959-916dc045-466e-4335-963f-7179dfb6072d.png)
Although both the used filter and the container size don't seem to have an effect. At least on the codepen examples, both render the same.
In that case, I'm kind of out of ideas.

**Comment by florian-strasser** [comment]
Applied to svg elements like rect it works pretty well, i nearly finished this :-) Just got some jiggly height when i drew the circle at the left or right of the box which i can't really understand yet

**Comment by florian-strasser** [comment]
I will send you my vue component when it's finished, maybe you can make use of it to have better crossbrowser comaptiblity

**Comment by thetarnav** [comment]
Thank you for the help :) You can make a PR if you want, but if you prefer to just send a file that is also fine. Generally, I think the SVG-based version of the component would work well as a separate component or when enabling it through a prop, like: `<GooeyBackground svg />`. Just to give it as an option, because I think there is still value in using HTML elements, for example: you can style the background to have more rounded corners. While svg-elements could provide different customization possibilities.

**Comment by florian-strasser** [comment]
Okay - i made alot of progress and for casual buttons it already working pretty well, but i need an circle option or eclipse as you called it :D The play Button is not calculating where the circle ends yet

But the effect it self is running smoothly in chrome, firefox and safari
http://vue.florian-strasser.de

**Comment by thetarnav** [comment]
The initial position of the blob is causing something like this:

 ![image](https://user-images.githubusercontent.com/24491503/142291910-cdb0c200-af15-465b-bc50-040f3f375865.png)

And I see that you have limited where the mouse movement gets detected, this is definitely good from a performance standpoint, but not as fun to the user, you barely get to experience that effect this way. This is why I went with a global event listener.
Everything else seems to be working nice. I had to inspect to actually tell that those are svgs, not HTML elements. Which is very promising.
